spec/controllers/api/v1/assignment_events_controller_spec.rb

```diff
@@ -46,6 +46,23 @@
       expect(assignment.reload.attributes).to eq original_attributes
     end
 
+    it "updates an existing assignment without a mixpanel_result" do
+      assignment = nil
+      Timecop.freeze(1.year.ago) do
+        assignment = FactoryBot.create(:assignment, visitor: visitor, split: split, variant: "control", mixpanel_result: nil)
+      end
+
+      assignment.reload
+
+      expect {
+        post :create, params: create_params
+      }.to change { PreviousAssignment.count }.by(1)
```

Comment:
It's surprising that `PreviousAssignment` would change. That means we're switching variants, which means that in the case of a changed `mixpanel_result`, we could move someone into a new variant. That shouldn't happen if they already have an assignment.

In the normal case, the window would be pretty small since the second assignment event call with the updated mixpanel result should occur very quickly. But that's not guaranteed.

I think this points to us not reusing `ArbitraryAssignmentCreation` just to update the `mixpanel_result`.